### DISCUSSION TITLE
New smooth sensitivity map for back optimisation

### DISCUSSION THREAD

**Discussion by liliande** [mdolab/dafoam#314] [General]
Hi,

Thanks for Dafoam, I'm using it to optimise the aerodynamics of an object and so to reduce drag.
Currently I'm working on the back of an object with Von Karman vortexes, boundary layer separation and so on. Unfortunately for that the openfoamv1812 uses unsmooth sensitivity maps and so the vortexes make some noises at the back. So Dafoam morphing results are not converging or not satisfying in these cases.

OpenFoamv2112 developed smooth sensitivity maps and it gives a more consistent map at the back like for the motorbike example.
![image](https://user-images.githubusercontent.com/13312931/171399153-f2c2a496-01e1-444c-b008-9568fb268eea.png)
So I would like to give smooth sensitivity maps to Dafoam at each / or at least at the first sensitivity function.

Do you think it is possible to import sensitivity maps from openfoamv2112 or pause Dafoam and replace openfoamv18012 maps by v2112 maps ?

Thanks for reading and have a nice day !

**Comment by friedenhe** [comment]
Yeah, unfortunately, if the flow has a large separation and the primal solver (e.g., simpleFoam) does not converge well (at least 6 orders of residual drop), DAFoam optimization won't have good progress.  Regarding your question, can you describe what exactly you are doing? DAFoam does not use the OpenFOAM's built-in continuous-adjoint-based sensitivity map feature during the optimization. Instead, it implements its own discrete adjoint to compute sensitivities. Also, it is not possible (at this moment) to replace OF-v1812 with OF-v2112 because we haven't AD OF-v2112 yet.

**Reply by liliande** [reply]
First, thanks for your interest for my question.
Ok I understand the problem between the two versions of OpenFoam. To be more clear, I will show you an example of what I would like to do with Dafoam on the motorbike, it can interest others people too I think.
I come back to you very soon.
